posthog/api/support_tickets.py

```diff
@@ -0,0 +1,343 @@
+import json
+from datetime import datetime
+
+from django.http import JsonResponse
+import requests
+from posthog.api.utils import get_token
+from posthog.exceptions import generate_exception_response
+from rest_framework.request import Request
+from rest_framework import status
+
+from posthog.models import User
+from posthog.models.team.team import Team
+from django.views.decorators.csrf import csrf_exempt
+
+from posthog.utils_cors import cors_response
+
+
+@csrf_exempt
+def get_support_tickets(request: Request):
+    token = get_token(None, request)
+
+    if not token:
+        return cors_response(
+            request,
+            generate_exception_response(
+                "support_tickets",
+                "API key not provided. You can find your project API key in PostHog project settings.",
+                type="authentication_error",
+                code="missing_api_key",
+                status_code=status.HTTP_401_UNAUTHORIZED,
+            ),
+        )
+
+    # TODO: fix organization_id & other fields don't exist in the cached object
+    # team = Team.objects.get_team_from_cache_or_token(token)
+    team = Team.objects.get(api_token=token)
+    if team is None:
+        return cors_response(
+            request,
+            generate_exception_response(
+                "support_tickets",
+                "Project API key invalid. You can find your project API key in PostHog project settings.",
+                type="authentication_error",
+                code="invalid_api_key",
+                status_code=status.HTTP_401_UNAUTHORIZED,
+            ),
+        )
+    zendesk_key = team.zendesk_key
+    if zendesk_key is None:
+        return cors_response(
+            request,
+            generate_exception_response(
+                "support_tickets",
+                "Zendesk API key not set. You can set it in PostHog project integrations settings.",
+                type="authentication_error",
+                code="missing_api_key",
+                status_code=status.HTTP_401_UNAUTHORIZED,  # probably should be some other error
+            ),
+        )
+
+    user = request.GET.get("user")
+    validation_token = request.GET.get("validation_token")
+    organization_id = team.organization_id
+
+        # do validation using the token above
+
+    headers = {"Authorization": f"Basic {zendesk_key}"}
+    response = requests.request(
+        "GET",
+        "https://posthoghelp.zendesk.com/api/v2/organizations/search",
+        data="",
+        headers=headers,
+        params={"external_id": organization_id},
```

Comment:
I don't think this will this work for any random team trying to use their zendesk as an integration, will it? For example for your personal site, with your own zendesk setup, you maybe wouldn't have external_id = org_id (posthog-internal detail). Shouldn't we instead be using the user param from the request?

Reply:
Would this work:
1. key is in posthog settings (like zendesk key) - we'd have "external_id"
2. we pass the value (organization_id for us) and secure that with the secret_token we have <- we can secure that the same way we would have secured user email? i.e. hogflix api will pass value and value_hash to posthog-js, which then passes both of these to posthog-api

Reply:
I'm super unclear about the zendesk API, but i'm assuming that it's not sufficient to just target by user, and we need this extra param?

If yes, then what you say is reasonable, we even don't need to secure it with the token, because securing one thing (user email) is sufficient for security, and also hashing this just creates annoyances for the user, they can just pass it in plain text ✅

Reply:
We essentially need to get the Zendesk org ID, in your case, we link it with the PostHog org ID (external ID).

The API key is something different again. But let's worry about productizing this after we get the demo working.

We'll probably need the users to give us a `giveZendeskOrgID()` function

Reply:
yes nothing for the demo agreed, so feel free to not read my comment.
There are a few things here:
1. It would be great if Hogflix developer could choose if they want the search to be by email or by org_id or by something else in Zendesk <- i.e. they control which tickets are visible to who
2. what they choose in (1) might be guessable, so we can't just verify the user - as I as a@b.c are the owner of that email, but I should not have access to org "Panda", but I can in js console send `getTickets(a@b.c, validation_token_for_a@b.c, "Panda") - there's no way for us to verify that a@b.c belongs to whatever "Panda" is in Zendesk, so instead we can do `getTickets("Panda", validation_token_for_panda)`